tests/TestCase.php

```diff
@@ -69,6 +69,49 @@ protected function assertDeprecated(callable $execution)
         $this->assertCount(1, $errors);
     }
 
+    /**
+     * Asserts that a document has expected values for some fields.
+     *
+     * Only fields in the expected document will be checked. The actual document
+     * may contain additional fields.
+     *
+     * @param array|object $expectedDocument
+     * @param array|object $actualDocument
+     */
+    protected function assertMatchesDocument($expectedDocument, $actualDocument)
+    {
+        $normalizedExpectedDocument = $this->normalizeBSON($expectedDocument);
+        $normalizedActualDocument = $this->normalizeBSON($actualDocument);
+
+        $extraKeys = [];
+
+        /* Avoid unsetting fields while we're iterating on the ArrayObject to
+         * work around https://bugs.php.net/bug.php?id=70246 */
```

Comment:
I had a great time discovering this bug. :stuck_out_tongue:

Reply:
For fun I would say: @sgolemon, fix yo shit ;-)